src/core/store/modules/vaults/vaults.actions.ts

```diff
@@ -456,12 +455,22 @@ const getVaultAllowance = createAsyncThunk<
 
   if (!userAddress) throw new Error('WALLET NOT CONNECTED');
 
-  return vaultService.getVaultAllowance({
+  const tokenAllowance = await vaultService.getVaultAllowance({
```

Comment:
I feel this sounds confusing, we are assigning the result of `getVaultAllowance()` to `tokenAllowance`?

Reply:
used that name since `TokenAllowance` is the type that is returned by this method. It is indeed a tokenAllowance. The service should be called `getVaultDepositAllowance` though imo. Since we will later add a different one for withdrawals.

Reply:
So not `getTokenAllowance()`?

Reply:
I would rename it to `getDepositAllowance` since its implicit it refers to vaults since its in the `vaultService`. That way its also aligned with sdk here https://github.com/yearn/yearn-sdk/blob/master/src/interfaces/vault.ts#L337. wdyt?

Reply:
That sounds good to me!

Reply:
refactored here :) https://github.com/yearn/yearn-finance-v3/pull/630/commits/e73ba66db30b3b4143aac516d62beae4ec40176b